crates/bevy_reflect/src/type_registry.rs

```diff
@@ -599,6 +599,23 @@ impl<T: Reflect> FromType<T> for ReflectFromPtr {
     }
 }
 
+/// In order to dynamic get component's dyn object.
+///
+/// This trait associates the dyn object corresponding to [`TypeData`] (because rust does not currently support `dyn T`. `T` is a generic parameter.),
+///
+/// This trait can be implemented automatically by the `#[reflect_trait]` macro,
+///
+/// If you implement [`TypeData`] for your component yourself, you should also implement this trait if you need to dynamically get the dyn object
```

Comment:
Again, we should avoid ECS terminology like "component". Maybe `type` is better?